node/node.go

```diff
@@ -1161,6 +1161,12 @@ func (node *AlgorandFullNode) VotingKeys(votingRound, keysRound basics.Round) []
 		}
 		participations = append(participations, part)
 		matchingAccountsKeys[part.Address()] = true
+
+		// This is usually a no-op, but the first time it will update the DB.
+		err := node.participationRegistry.Register(part.ParticipationID(), keysRound)
```

Comment:
please benchmark the `Register` and make sure it won't take too long. ( if it takes less than 50ms, then it's probably just fine. Otherwise, we need to consider moving it to a go-routine )

Reply:
Looks very fast to me. When you say ms are you think microseconds or milliseconds? Either way we're still under 50 if there's only 1 account registered:
```
BenchmarkKeyRegistration1/KeyInsert_1
BenchmarkKeyRegistration1/KeyInsert_1-8    	  202903	      6154 ns/op
BenchmarkKeyRegistration1/KeyRegistered_1
BenchmarkKeyRegistration1/KeyRegistered_1-8         	   26833	     44209 ns/op
BenchmarkKeyRegistration1/NoOp_1
BenchmarkKeyRegistration1/NoOp_1-8                  	  194818	      5980 ns/op
BenchmarkKeyRegistration5
BenchmarkKeyRegistration5/KeyInsert_5
BenchmarkKeyRegistration5/KeyInsert_5-8             	   41196	     30087 ns/op
BenchmarkKeyRegistration5/KeyRegistered_5
BenchmarkKeyRegistration5/KeyRegistered_5-8         	    5782	    221874 ns/op
BenchmarkKeyRegistration5/NoOp_5
BenchmarkKeyRegistration5/NoOp_5-8                  	   39292	     29898 ns/op
BenchmarkKeyRegistration10
BenchmarkKeyRegistration10/KeyInsert_10
BenchmarkKeyRegistration10/KeyInsert_10-8           	   19551	     59042 ns/op
BenchmarkKeyRegistration10/KeyRegistered_10
BenchmarkKeyRegistration10/KeyRegistered_10-8       	    2823	    448233 ns/op
BenchmarkKeyRegistration10/NoOp_10
BenchmarkKeyRegistration10/NoOp_10-8                	   19406	     60828 ns/op
BenchmarkKeyRegistration50
BenchmarkKeyRegistration50/KeyInsert_50
BenchmarkKeyRegistration50/KeyInsert_50-8           	    3784	    394140 ns/op
BenchmarkKeyRegistration50/KeyRegistered_50
BenchmarkKeyRegistration50/KeyRegistered_50-8       	     394	   3207132 ns/op
BenchmarkKeyRegistration50/NoOp_50
BenchmarkKeyRegistration50/NoOp_50-8                	    2625	    427918 ns/op
```

Reply:
milliseconds. and these looks good. ( I just wanted to make sure )

Reply:
Do you think it's worth making this asynchronous? This design means that we'll frequently re-check for keys to register so it no longer needs to update the DB in real-time.

Reply:
The `VotingKeys` is called by the pseudonode once every round so that it could use the keys for voting on that round. By slowing this down, we're holding off the agreement from making the proposal message ( which is a time critical). This could be really bad if it takes too long; however, if we're taking about 50Kns, than it's probably wouldn't make a difference. Just a side note that if we take any lock in there - that lock might take considerably more than 50Kns, so we need to be careful around that.

Reply:
There is a lock taken for most of the record function. There is probably a clever solution with multiple mutexes that would avoid a DB access lock in the agreement path, I'll need to think about that more.

Reply:
I'll create a follow-up ticket for avoiding mutexes and DB operations in Record/Register.